File: hw/villagomez/u1/hw10_ChickenFile/ChickenFarmSimulator2/src/ec/edu/espe/farmsilumator/view/ChickenFarmSimulator.java
package ec.edu.espe.farmsilumator.view;

import ec.edu.espe.farmsilumator.controller.SaveChickenData;
import ec.edu.espe.farmsilumator.model.Chicken;
import java.util.Date;
import java.util.Scanner;

/**
 *
 * @author Doménica Villagómez, CommitCrew, DCCO-ESPE
 */
public class ChickenFarmSimulator {
    public static void main(String[] args) {
        System.out.println("Domenica's Chicken Farm Simulator V. 0.5");
        
        try(Scanner input = new Scanner(System.in)) {
            
            int id;
            String name;
            String color;
            int age;
            boolean molting;
            Date bornOnDate;

            System.out.println("Enter the ID:");
            String idStr = input.nextLine();
            id = Integer.parseInt(idStr);

            System.out.println("Enter the name:");
            name = input.nextLine();

            System.out.println("Enter the color:");
            color = input.nextLine();

            System.out.println("Enter the age:");
            String ageStr = input.nextLine();
            age = Integer.parseInt(ageStr);

            System.out.println("Enter if its molting:");
            String moltingStr = input.nextLine();
            molting = Boolean.parseBoolean(moltingStr);

            bornOnDate = new Date();

            Chicken chicken = new Chicken(id, name, color, age, molting, bornOnDate);

            SaveChickenData.saveChickenData(chicken);

            System.out.println("Chicken --> \n");
            
        }
            
    }
        

    public ChickenFarmSimulator() {
    }
}

File: hw/villagomez/u1/hw10_ChickenFile/ChickenFarmSimulator2JS/ec.edu.espe.chickenfarmsimulator.view/ChickenFarmSimulator.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

const SaveChickenData = require('./ec.edu.espe.chickenfarmsimulator.controller/SaveChickenData'); 
const Chicken = require('./ec.edu.espe.chickenfarmsimulator.model/Chicken');

class ChickenFarmSimulator {
    static main() {
        console.log("Domenica's Chicken Farm Simulator V. 0.5");

        const askUserForChickenData = async () => {
            const id = await new Promise((resolve) => readline.question('Enter the ID: ', resolve));
            const name = await new Promise((resolve) => readline.question('Enter the name: ', resolve));
            const color = await new Promise((resolve) => readline.question('Enter the color: ', resolve));
            const age = await new Promise((resolve) => readline.question('Enter the age: ', resolve));
            const moltingStr = await new Promise((resolve) => readline.question('Enter if its molting (true/false): ', resolve));
            const molting = moltingStr.toLowerCase() === 'true';
            const bornOnDate = new Date();

            return new Chicken(parseInt(id), name, color, parseInt(age), molting, bornOnDate);
        };

        (async () => {
            try {
                const chicken = await askUserForChickenData();
                const filePath = "./data/chickenData2.txt"; // Ruta relativa a la carpeta del archivo
                const result = await SaveChickenData.saveChickenData(chicken, filePath);
                console.log(result);
                console.log('Chicken --> \n', chicken);
            } catch (error) {
                console.error('Error:', error.message);
            } finally {
                readline.close();
            }
        })();
    }
}

module.exports = ChickenFarmSimulator;